.github/workflows/lean_doc.yml

```
@@ -0,0 +1,90 @@
+on:
+  push:
+
+jobs:
+  build_docs:
+    runs-on: ubuntu-latest
+    name: Build docs
+    steps:
+
+    - name: 📁 Checkout project
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+
+    - name: 🔧 Install elan
+      run: |
+        set -o pipefail
+        curl https://raw.githubusercontent.com/leanprover/elan/master/elan-init.sh -sSf | sh -s -- --default-toolchain none -y
+        ~/.elan/bin/lean --version
+        echo "$HOME/.elan/bin" >> $GITHUB_PATH
+
+    - name: 🐍 Install Python
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.8
+
+    - name: 📁 Checkout Lean's doc-gen
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+        repository: leanprover-community/doc-gen
+        path: doc-gen
+
+    - name: 🐍 Install Python dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r doc-gen/requirements.txt
+        ls doc-gen
+
+    - name: ✏️ Mangle paths for doc-gen
+      run: |
+        lean_version="$(sed '/^lean_version/!d;s/.*"\(.*\)".*/\1/' leanpkg.toml)"
+        mathlib_git_hash="$(sed '/rev/!d;s/.*"\([a-z0-9]*\)".*/\1/' leanpkg.toml)"
+        cd doc-gen
+        sed -i "s/rev = \"\S*\"/rev = \"$mathlib_git_hash\"/" leanpkg.toml
+        
+        # Force doc_gen project to match the Lean version used in CI.
+        # If they are incompatible, something in doc_gen will fail to compile,
+        # but this is better than trying to recompile all of mathlib.
+        elan override set "$lean_version"
+
+        leanproject get-mathlib-cache
+        echo -e "path ../src" >> leanpkg.path
+        cat leanpkg.path
+        (cd ../src;
+        find "." -name \*.lean -not -name all.lean \
+          | sed 's,^\./,,;s,\.lean$,»,;s,/,».«,g;s,^,import «,' \
+          | sort >./all.lean);
+        
+        mkdir ../docs
+        cp _target/deps/mathlib/docs/references.bib ../docs
+
+        cat ../src/all.lean
+        lean --path
+
+    - name: 📝 Export Lean symbol information
+      run: |
+        cd doc-gen
+        lean -v
+        # the json and diagnostics are emitted on stdout, but the json is always the last (very long) line.
+        lean src/entrypoint.lean > export.json || true
```

```suggestion
        lean --run src/entrypoint.lean > export.json || true
```
per https://github.com/leanprover-community/doc-gen/commit/bff41dc8b3e11e5d8d1f44422e4f364de4e20c57

---

I think that's now overkill: this is what I use elsewhere:

https://github.com/pygae/lean-ga/blob/6268eb9b664d99e8c1e0bc0033c8135e29734857/.github/workflows/lean_doc.yml#L68-L77